classes/admin.php

```diff
@@ -170,11 +170,11 @@ public static function adminLanguages()
 
         /** @var \DirectoryIterator $directory */
         foreach (new \DirectoryIterator($path) as $file) {
-            if ($file->isDir() || $file->isDot() || Utils::startsWith($file->getBasename(), '.')) {
+            if ($file->isDir() || $file->isDot() || Utils::startsWith($file->getFilename(), '.')) {
                 continue;
             }
 
-            $lang = basename($file->getBasename(), '.yaml');
+            $lang = $file->getBasename('.yaml');
```

Comment:
Didn't you say that the method is not locale friendly? How is this better than `basename($file->getFilename(), '.yaml');`?

Reply:
The reason that `DirectoryIterator::getBasename` and `SplFileInfo::getBasename` are not locale friendly is because they call `basename` underneath (See [implementation](https://github.com/php/php-src/blob/master/ext/spl/spl_directory.c#L997-L1041)). In other words

```php
(basename($file->getBasename(), '.yaml') === $file->getBasename('.yaml')) &&
($file->getBasename('.yaml') === basename($file->getFilename(), '.yaml'))
```

Reply:
It also matches up with the usage of `DirectoryIterator::getBasename` in the main grav repository (See [here](https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Data/Blueprints.php#L66-L73) and [here](https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Themes.php#L302-L308))

Reply:
@mahagr, does this answer your question?